xo-install.sh

```diff
@@ -573,13 +573,16 @@ function PrepInstall {
     fi
 
     echo
-    # keep the actual source code in one directory and either clone or git pull depending on if directory exists already
+    # keep the actual source code in one directory and either clone or git fetch depending on if directory exists already
     printinfo "Fetching $XO_SVC_DESC source code"
     if [[ ! -d "$XO_SRC_DIR" ]]; then
         runcmd "mkdir -p \"$XO_SRC_DIR\""
         runcmd "git clone \"${REPOSITORY}\" \"$XO_SRC_DIR\""
     else
-        runcmd "cd \"$XO_SRC_DIR\" && git pull --ff-only"
+        runcmd "cd \"$XO_SRC_DIR\" && git remote set-url origin \"${REPOSITORY}\""
```

Comment:
Since runcmd runs in a subshell, you need to change directory for each individual git command, otherwise they'll run inside same directory where xo-install.sh is invoked, which most likely is XenOrchestraInstallerUpdater directory and causes script to fail as it wipes logs, config files etc :) yeah i know there's the cd command you didn't edit at the end, it's kinda obsolete and relic from the past when commands weren't run by separate function (should clean up at some point).


You could also group all those commands together as they rely on each other anyway:


```
runcmd "cd \"$XO_SRC_DIR\" && git remote set-url origin \"${REPOSITORY}\" && \
    git fetch --prune && \
    git reset --hard origin/master && \
    git clean -xdff"
```

Reply:
Good point, done